mne_connectivity/spectral/time.py

```diff
@@ -631,3 +600,25 @@ def _foi_average(conn, foi_idx):
         f_e += 1 if f_s == f_e else f_e
         conn_f[..., n_f, :] = conn[..., f_s:f_e, :].mean(-2)
     return conn_f
+
+
+def _compute_freqs(n_times, sfreq, freqs, mode):
+    from scipy.fft import rfftfreq
+    # get frequencies of interest for the different modes
+    if freqs is not None:
+        if any(freqs > (sfreq / 2.)):
```

Comment:
I think this is defined already in `spectral.epochs.py`. Can we use that one?

Reply:
Can you add `freqs` into the existing function and then use it? 

Having one private function makes the code more robust, easier to maintain, and less likely to introduce runtime bugs in the future.

Reply:
Technically yes, but then we would have to prevent the use of `cwt_freqs` together with `multitaper` in `spectral_connectivity_epochs`. In that case `freqs` must be equal to `rfftfreq(n_times, 1./sfreq)` as that's the shape of the output of `_csd_from_mt` and `_psd_from_mt`.

Reply:
I think that is fine. So, we would check `if freqs is not None and mode == 'multitaper'` and raise an error.

The function is essentially doing the same thing regardless of which one it's being called from.

Reply:
This should be the case only in `spectral_connectivity_epochs` (where multitaper frequencies must be defined by `rfftfreq` for the method to work). In `spectral_connectivity_time` there isn't really much reason to even use `rfftfreq` to obtain frequencies as it uses a different (time-resolved) multitaper implementation. If we require `freqs` anyway with `morlet`, it might make sense to go back to requiring them for `multitaper` as well given that, as far as I understand, in the time-resolved `tfr` implementation the difference between `morlet` and `multitaper` is only in the type of wavelet used (+ multiple tapers with multitaper).

Reply:
Maybe a solution could be to add the connectivity paradigm (epochs vs time) to the parameters of `_compute_freqs`. Although the reasoning in my previous comment would still favor requiring `freqs` in Multitaper mode as well.

Reply:
Sure, I think requiring `freqs` could be fine. I'm not 100% sure I'm following wrt all the different possible combinations of parameters here, so I can take a look after you push up the proposed change?